packages/ui-components/src/components/toolbar.tsx

```diff
@@ -607,10 +625,57 @@ export class ToolbarButton extends ReactWidget {
   constructor(private props: ToolbarButtonComponent.IProps = {}) {
     super();
     addToolbarButtonClass(this);
+    this._enabled = props.enabled ?? true;
+    this._pressed = this._enabled === true ? props.pressed ?? false : false;
+    this._onClick = props.onClick;
   }
+
+  set pressed(value: boolean) {
+    if (value != this._pressed) {
+      this._pressed = value;
+      this.update();
+    }
+  }
+
+  get pressed() {
+    return this._enabled === true ? this._pressed : false;
+  }
+
+  set enabled(value: boolean) {
+    if (value != this._enabled) {
+      this._enabled = value;
```

Comment:
You should enforce in setter that pressed cannot be set if the button is disabled. So the getter should point to the private value only (otherwise in further use, `this._pressed` could be different that `this.pressed`).

```suggestion
  set pressed(value: boolean) {
    if (this.enabled && value !== this._pressed) {
      this._pressed = value;
      this.update();
    }
  }

  get pressed() {
    return this._pressed;
  }

  set enabled(value: boolean) {
    if (value !== this._enabled) {
      this._enabled = value;
      if (!this._enabled) {
        this._pressed = false;
      }
```